service/history/workflow/workflow_task_state_machine.go

```diff
@@ -1087,11 +1080,64 @@ func (m *workflowTaskStateMachine) afterAddWorkflowTaskCompletedEvent(
 	attrs := event.GetWorkflowTaskCompletedEventAttributes()
 	m.ms.executionInfo.LastCompletedWorkflowTaskStartedEventId = attrs.GetStartedEventId()
 	m.ms.executionInfo.MostRecentWorkerVersionStamp = attrs.GetWorkerVersion()
-	if err := m.ms.CompleteDeploymentTransition(attrs.GetVersioningBehavior()); err != nil {
-		return err
+
+	taskDeployment := attrs.GetDeployment()
+	taskBehavior := attrs.GetVersioningBehavior()
```

Comment:
```suggestion
	wftDeployment := attrs.GetDeployment()
	wftBehavior := attrs.GetVersioningBehavior()
```
same reason: I would avoid word "task" everywhere where possible.